### PR TITLE
[ CHEC-753] Adding inline alert version

### PR DESCRIPTION
**Pull request by kvisca**
- Adding the inline alert version.
- Adding the new square icons.
- Adding option to disable close button
- Updating story to reflect new options and inline versions.  

Screen Shot 2020-07-02 at 5 47 08 PM

### CODE REVIEW

File: src/stories/components/BaseAlert.stories.mdx [hunk 1]
```diff
@@ -93,4 +103,77 @@ import BaseAlert from '../../components/BaseAlert.vue';
...
+          if (variant == 'error') {
+            return 'close-square';
+          }
+          if (variant == 'warning') {
+            return 'exclamation-square';
+          }
+          if (variant == 'info') {
+            return 'info-square';
+          }
+          return 'info-square';
```
Comment (ScopeyNZ) [inline src/stories/components/BaseAlert.stories.mdx:151]: What do you think about making this stuff part of the component - as a `show-icon` prop?

File: src/components/BaseAlert.vue [hunk 2]
```diff
@@ -54,21 +62,25 @@ export default {
...
+    svg {
+      @apply inline h-4 w-4;
+    }
   }
 
   &__close-icon {
     @apply ml-4 p-2 bg-transparent rounded text-white cursor-pointer outline-none;
-
+    margin-top: -2rem;
+    margin-bottom: -2rem;
```
[the remark this answers is not in the document]
Comment (ScopeyNZ) [reply src/components/BaseAlert.vue:83]: There's `@apply -my-8` for this too.

File: src/components/BaseAlert.vue [hunk 3]
```diff
@@ -89,6 +101,55 @@ export default {
...
+    &.alert--error{
+      @apply bg-red-100;
+      .alert__text{
+        svg{
+          @apply text-red-500
+        }
+      }
+
+    }
+    &.alert--warning{
```
[the remark this answers is not in the document]
Comment (ScopeyNZ) [reply src/components/BaseAlert.vue]: See https://github.com/chec/ui-library/blob/ff95f6044827e3e1b163b1efa30b6774aa1553d2/src/components/BaseButton.vue#L239

File: src/components/BaseAlert.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (kvisca) [reply src/components/BaseAlert.vue:83]: This is so when the close button is removed the container doesn't become smaller than it should be.

**Comment by robbieaverill** [comment]
Can you remove the `.DS_Store` file and add that filename to your global gitignore rules?

File: src/components/BaseAlert.vue [hunk 4]
```diff
@@ -35,10 +38,37 @@ export default {
...
+    disableClose: Boolean,
+    /**
+     * Show the alert icon.
+     */
+    showIcon: Boolean,
   },
   computed: {
     classObject() {
-      return `alert--${this.variant}`;
+      return `alert--${this.variant} ${this.inline ? 'alert--inline' : ''} ${this.icon ? 'alert--icon' : ''}`;
```
Comment (ScopeyNZ) [inline src/components/BaseAlert.vue:56]: This is fine - but for future reference you can write this as:

```suggestion
      return [`alert--${this.variant}`, {
        'alert--inline': this.inline,
        'alert--icon': this.icon,
      }];
```

I think it's a tiny bit more readable.

Comment (kvisca) [reply src/components/BaseAlert.vue]: Updated

**Comment by ScopeyNZ** [comment]
Nice. I was just about to merge but I remembered that we actually have the non-coloured variant in use on the new docs:

https://www.figma.com/file/ypYnoIU8yJfnhek3h2ZRvQ/Docs-Website?node-id=3%3A686

We'll need that as an option

Additionally, can you add a knob for the "inline" prop to the "Default" story?

**Comment by kvisca** [comment]
> Additionally, can you add a knob for the "inline" prop to the "Default" story?

Fixed the issue where info displayed in incorrect color and updated the story.

**Comment by ScopeyNZ** [comment]
🎉

**Pull request merged**